### PR TITLE
NKAI: namespace

### PR DESCRIPTION
**Pull request by nullkiller**
Nothing interesting. Just added a namespace for Android as it builds everything in a single lib

### CODE REVIEW

File: AI/Nullkiller/Goals/Win.cpp [hunk 1]
```diff
@@ -1,3 +1,6 @@
+namespace Nullkiller
+{
+
 /*
 * Win.cpp, part of VCMI engine
 *
```
Comment (kambala-decapitator) [inline AI/Nullkiller/Goals/Win.cpp:6]: please place after the header

File: AI/Nullkiller/Goals/Win.h [hunk 2]
```diff
@@ -1,3 +1,6 @@
+namespace Nullkiller
+{
+
 /*
 * Win.h, part of VCMI engine
 *
```
Comment (kambala-decapitator) [inline AI/Nullkiller/Goals/Win.h:6]: please place after the header

File: AI/Nullkiller/Pathfinding/AINodeStorage.h [hunk 3]
```diff
@@ -10,10 +10,13 @@
 
 #pragma once
 
+namespace NKAI
+{
 #define PATHFINDER_TRACE_LEVEL 0
 #define AI_TRACE_LEVEL 0
 #define SCOUT_TURN_DISTANCE_LIMIT 3
 #define MAIN_TURN_DISTANCE_LIMIT 5
+}
```
Comment (kambala-decapitator) [inline AI/Nullkiller/Pathfinding/AINodeStorage.h:19]: macros don't care about namespace iirc

File: AI/gen-ns.ps1 [hunk 4]
```diff
@@ -0,0 +1,13 @@
+$files = Get-ChildItem -Path .\Nullkiller\ -Recurse -ErrorAction SilentlyContinue -Force -Filter '*.cpp'
```
Comment (kambala-decapitator) [inline AI/gen-ns.ps1:1]: should this file really be in the repo?

File: AI/Nullkiller/AIGateway.cpp [hunk 5]
```diff
@@ -23,8 +23,15 @@
 #include "AIGateway.h"
 #include "Goals/Goals.h"
 
+VCMI_LIB_NAMESPACE_BEGIN
+
 class CGVisitableOPW;
 
+VCMI_LIB_NAMESPACE_END
```
Comment (kambala-decapitator) [inline AI/Nullkiller/AIGateway.cpp:30]: not sure why this is needed - `using namespace` is already in StdInc.h

File: AI/Nullkiller/AIUtility.h [hunk 6]
```diff
@@ -48,19 +48,25 @@
...
-class CCallback;
-class Nullkiller;
-struct creInfo;
+VCMI_LIB_NAMESPACE_BEGIN
 
 typedef const int3 & crint3;
 typedef const std::string & crstring;
 typedef std::pair<ui32, std::vector<CreatureID>> dwellingContent;
 
+VCMI_LIB_NAMESPACE_END
```
Comment (kambala-decapitator) [inline AI/Nullkiller/AIUtility.h:63]: not sure why this is needed - `using namespace` is already in StdInc.h

**Review by kambala-decapitator — CHANGES_REQUESTED** [review]

File: AI/gen-ns.ps1 (hunk 4 above)
Comment (nullkiller) [reply AI/gen-ns.ps1:1]: Wow, will remove)

File: AI/Nullkiller/Behaviors/Behavior.cpp [hunk 7]
```diff
@@ -0,0 +1,5 @@
+namespace Nullkiller
+{
+
+
+}
```
Comment (kambala-decapitator) [inline AI/Nullkiller/Behaviors/Behavior.cpp:5]: interesting file :)

File: AI/Nullkiller/AIUtility.h (hunk 6 above)
Comment (kambala-decapitator) [reply AI/Nullkiller/AIUtility.h]: oh, it's a header - here it's definitely needed

File: AI/Nullkiller/Behaviors/Behavior.cpp (hunk 7 above)
Comment (nullkiller) [reply AI/Nullkiller/Behaviors/Behavior.cpp:5]: Ha-ha. There was some idea) I think it is also not under cmakelist.txt

File: AI/Nullkiller/Goals/Win.cpp (hunk 1 above)
Comment (nullkiller) [reply AI/Nullkiller/Goals/Win.cpp]: Removed file

File: AI/Nullkiller/Goals/Win.h (hunk 2 above)
Comment (nullkiller) [reply AI/Nullkiller/Goals/Win.h]: Removed file

File: AI/Nullkiller/Pathfinding/AINodeStorage.h (hunk 3 above)
Comment (nullkiller) [reply AI/Nullkiller/Pathfinding/AINodeStorage.h]: Renamed defines to have NKAI prefix

File: AI/Nullkiller/AIGateway.cpp (hunk 5 above)
Comment (nullkiller) [reply AI/Nullkiller/AIGateway.cpp]: Cleared some unused forward declarations

File: AI/Nullkiller/Behaviors/Behavior.cpp (hunk 7 above)
Comment (nullkiller) [reply AI/Nullkiller/Behaviors/Behavior.cpp]: Removed

File: AI/Nullkiller/AIUtility.h (hunk 6 above)
Comment (nullkiller) [reply AI/Nullkiller/AIUtility.h]: Refactored a bit

**Review by kambala-decapitator — APPROVED** [review]

**Pull request merged**